tests/unit/src/main/scala/tests/BaseAmmoniteSuite.scala

```diff
@@ -0,0 +1,238 @@
+package tests
+
+import org.eclipse.lsp4j.MessageActionItem
+import scala.meta.internal.metals.Messages
+
+abstract class BaseAmmoniteSuite(scalaVersion: String)
+    extends BaseLspSuite("ammonite") {
+
+  override def munitIgnore: Boolean =
+    !isValidScalaVersionForEnv(scalaVersion) ||
+      // temporary, remove once Ammonite is available for 2.12.11
+      scalaVersion == "2.12.11"
+
+  override def newServer(workspaceName: String): Unit = {
+    super.newServer(workspaceName)
+    server.client.showMessageRequestHandler = { params =>
+      if (params == Messages.ImportAmmoniteScript.params())
+        Some(new MessageActionItem(Messages.ImportAmmoniteScript.dismiss))
+      else
+        None
+    }
+  }
+
+  test("simple script") {
+    // single script with import $ivy-s
+    for {
+      _ <- server.initialize(
+        s"""
+           |/metals.json
+           |{
+           |  "a": {
+           |    "scalaVersion": "$scalaVersion"
+           |  }
+           |}
+           |/main.sc
+           | // scala $scalaVersion
+           |import $$ivy.`io.circe::circe-core:0.12.3`
+           |import $$ivy.`io.circe::circe-generic:0.12.3`
+           |import $$ivy.`io.circe::circe-parser:0.12.3`
+           |import io.circe._, io.circe.generic.auto._, io.circe.parser._, io.circe.syntax._
+           |
+           |sealed trait Foo
+           |case class Bar(xs: Vector[String]) extends Foo
+           |case class Qux(i: Int, d: Option[Double]) extends Foo
+           |
+           |val foo: Foo = Qux(13, Some(14.0))
+           |
+           |val json = foo.asJson.noSpaces
+           |
+           |val decodedFoo = decode[Foo](json)
+           |""".stripMargin
+      )
+      _ <- server.didOpen("main.sc")
+      _ <- server.didSave("main.sc")(identity)
+      _ <- server.executeCommand("ammonite-start")
+
+      // asJson
+      // via Ammonite-generated Semantic DB
+      _ <- server.assertDefinitionAtLocation(
+        "main.sc",
+        12,
+        18,
+        ".metals/readonly/io/circe/syntax/package.scala"
```

Comment:
Maybe we can include those as comments in the file? Similar to how we do a lot of the expected suites like `DefinitionLspSuite`?

Reply:
I slightly changed `assertDefinitionAtLocation`, so that it infers the exact line / char itself. That should make the resulting code more maintainable.

I didn't inline that like in `DefinitionLspSuite`, for fear the resulting output would change in the future (there are still some definitions, like `import $file.foo, foo.Something`, that don't work as expected yet).